### PR TITLE
Publish Pub/Sub messages when scaling Spanner instances

### PR DESCRIPTION
**Pull request by GarupanOjisan**
This change introduces a feature that publishes Pub/Sub messages whenever Spanner instances are scaled up/down.
This enables other services to respond to the changes in Spanner capacity.

This feature addresses the following issue.
It allows users to subscribe to the Pub/Sub messages and send notifications to any platforms.

https://github.com/cloudspannerecosystem/autoscaler/issues/12

### CODE REVIEW

**Comment by GarupanOjisan** [comment]
This PR resolves issue #12

**Review by henrybell — CHANGES_REQUESTED** [review]
I added a couple of comments as well as the two additional commits as discussed:

- [Update working directory to core](https://github.com/cloudspannerecosystem/autoscaler/pull/91/commits/e0a0354cb5d8ea52f7398bdfeaa4870b990476b9)
- [Add 'downstream' identifier to loglines](https://github.com/cloudspannerecosystem/autoscaler/pull/91/commits/2aedb80f6decdaa40e42f8c8a19659d0e0c6b9af)

I tested with Cloud Functions and GKE -- in both cases the downstream messages were successfully sent follow scaling events.

File: poller/poller-core/index.js [hunk 1]
```diff
@@ -208,8 +208,8 @@ function postPubSubMessage(spanner, metrics) {
 
   return topic.publish(messageBuffer)
       .then(
-          log(`----- Published message to topic: ${spanner.scalerPubSubTopic}`,
-            {severity: 'INFO', projectId: spanner.projectId, instanceId: spanner.instanceId, payload: spanner}))
+          log(`Published message to topic: ${spanner.scalerPubSubTopic}`,
+              'INFO', spanner))
```
[the remark this answers is not in the document]
Comment (GarupanOjisan) [reply poller/poller-core/index.js:212]: Of course, I didn't consider that fully. We'll keep it.

File: scaler/scaler-core/index.js [hunk 2]
```diff
@@ -151,6 +167,7 @@ async function processScalingRequest(spanner, autoscalerState) {
     try {
       await scaleSpannerInstance(spanner, suggestedSize);
       await autoscalerState.set();
+      await publishDownstreamEvent('SCALING', spanner, suggestedSize);
```
[the remark this answers is not in the document]
Comment (GarupanOjisan) [reply scaler/scaler-core/index.js:170]: I would assume that what the user wants to know is the actual success or failure of the scale. At the very least, we need to be able to distinguish between a failure of a publishDownstreamEvent and an actual scale failure.

**Review by henrybell — APPROVED** [review]
LGTM! Many thanks for your work on this PR.

**Pull request merged**